src/plot_api/plot_api.js

```diff
@@ -1586,7 +1586,9 @@ Plotly.restyle = function restyle(gd, astr, val, traces) {
         'outsidetextfont.size', 'outsidetextfont.family', 'outsidetextfont.color',
         'hole', 'scalegroup', 'domain', 'domain.x', 'domain.y',
         'domain.x[0]', 'domain.x[1]', 'domain.y[0]', 'domain.y[1]',
-        'tilt', 'tiltaxis', 'depth', 'direction', 'rotation', 'pull'
+        'tilt', 'tiltaxis', 'depth', 'direction', 'rotation', 'pull',
```

Comment:
@etpinard just thinking about how these arrays could simply be done by flags in the `attribute.js` definitions but it's not trivial due to the irregularities:

1. Sometimes (as `domain.x[0]` here) the check is below the prop level (this is easy to cover though)
2. There are multiple sets: `recalcAttrs`, `autorangeAttrs`, `replotAttrs`, 'swapAttrs' (still easy to cover but not pretty)
3. Lots of manual overrides (see `elseif` tree with `doextra`s: https://github.com/plotly/plotly.js/blob/master/src/plot_api/plot_api.js#L1736-L1814
4. Innumerable special casing for a wide diversity of things such as `pie`, `LAYOUT`, `if(ai === 'orientationaxes')` etc.

It's possible to first cover 1, 2 and maybe 3 but even that leaves quite a lot of entanglement.